File: go/src/prime_factors/prime_factors.go
package prime_factors

func Factors(num int) []int {
    return getFactors(num, 2);
}

func getFactors(num int, can int) []int {
    if can > num {
        return []int{}
    } else if (num % can == 0) {
        return append([]int{can}, getFactors((num / can), can)...)
    } else {
        return getFactors(num, can + 1)
    }
}

File: scala/src/main/scala/PrimeFactors.scala
package com.crebma.primeFactors

object PrimeFactors {

    def isEven(num: Int) : Boolean = {
        num % 2 == 0
    }

    def isGreaterThanTwo(num: Int) : Boolean = {
        num > 2
    }

    def isOne(num: Int) : Boolean = {
        num == 1
    }

    def factors(num: Int, can: Int = 2) : List[Int] = {
        if (can > num) {
            List[Int]()
        } else if (isEven(num) && isGreaterThanTwo(num)) {
            List[Int](2) ++ factors(num/2)
        } else {
            List[Int](num)
        }
    }
}
